dms/models/storage.py

```diff
@@ -85,6 +85,12 @@ class Storage(models.Model):
     )
 
     model_ids = fields.Many2many("ir.model", string="Linked Models")
+    inherit_access_from_parent_record = fields.Boolean(
+        string="Inherit access from parent record", default=False
+    )
+    attachment_write_need_check = fields.Boolean(
+        string="Attachment write need check", default=False
+    )
```

Comment:
Please add:

1. Help attributes on both fields.
2. An onchange that sets `inherit_access_from_parent_record = False` when `save_type != 'attachment'` (this should make all my other comments actually work).

Also, I think `attachment_write_need_check` is not a very helpful name or logic. What the user cares about is usually the fact that the attachment comes from a chatter message or that it was attached manually to the record, right? That's what the name should do (and it should be the actual implementation) IIUC.

A standard human being doesn't know that Odoo calls `write` the act of editing a record, and even less than every attachment coming from an email goes through that. Let's make this more friendly. 😃